pkg/kubelet/kubelet.go

```diff
@@ -2703,7 +2703,7 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod) {
 			//TODO(vinaykul,InPlacePodVerticalScaling): Can we recover from this in some way? Investigate
 			klog.ErrorS(err, "SetPodAllocation failed", "pod", klog.KObj(pod))
 		}
-		*pod = *updatedPod
+		pod = updatedPod
 	}
 	if resizeStatus != "" {
 		// Save resize decision to checkpoint
```

Comment:
It ends up doing nothing (resize fails) last I tried simulating failure. I updated PR https://github.com/kubernetes/kubernetes/pull/116702 to explicitly return for now so it is more obvious. There's a TODO to investigate if this can be handled better. I have some ideas but I need to really understand Clayton's plans to see if we can work something out to make it more resilient.
/cc @smarterclayton